Joystream/joystream-api-examples
# Nicaea: Add scripting

Adding a simple way to run javascript "scripts", the same way you can run them in the pioneer javascript toolbox: https://testnet.joystream.org/#/js

Included scripts for exporting and injecting data objects, from: https://github.com/Joystream/joystream/pull/877#issuecomment-653601853

To test for Nicaea, run dev chain then:

```bash
yarn && yarn build
yarn script injectDataObjects
```

after injection is complete...

`yarn script listDataDirectory`

To see list of scripts available:

`yarn script`

The injectDataObjects script requires sudo accout, on a dev chain this is derived and available in keyring from `//Alice` SURI which is automatically added.

Optionally you can pass as last argument after script name a single SURI to be added to the keyring.

`yarn script injectDataDirectory //mySeeD12`
`yarn script injectDataDirectory "my mneumonic seed words..."`
`yarn script injectDataDirectory 0x123456...private.key.....789`

If a script attempts to sign a tx with an account not in the keyring it will fail with: `Error: Unable to retrieve keypair`

Note: in pioneer a keyring is only passed to the script if a development chain is detected (for obvious security reasons)

Improvements that can be made:

In the node environment we always pass the keyring (because I didn't implement a custom `Signer` when creating the `ApiPromise`) so we have to write some conditional code in the script for now.
